src/base/memory_monitor_sr.hpp

```diff
@@ -0,0 +1,266 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * memory_monitor_sr.hpp - memory monitoring module header
+ */
+
+#ifndef _MEMORY_MONITOR_HPP_
+#define _MEMORY_MONITOR_HPP_
+
+#include "perf_def.hpp"
+//#include "thread_compat.hpp"
+#include "memory_monitor_common.h"
+
+//#ifdef __cplusplus
+#include <cstring>
+#include <type_traits>
+#include <cstdint>
+#include <atomic>
+#include <vector>
+
+#include <cassert>
+
+#define MMM_PARSE_MASK 0x0000FFFF
+#define MMM_MAKE_STAT(module_idx) ((module_idx) << 16)
+#define MMM_MODULE_IDX_FROM_STAT(stat) ((stat) >> 16)
+#define MMM_COMP_INFO_IDX_FROM_STAT((stat) & MMM_PARSE_MASK)
+
+#ifdef SERVER_MODE
+typedef enum
+{
+  HEAP = MMM_MAKE_STAT (HEAP_MODULE),
+  HEAP_CLASSREPR,
+  MMM_STAT_END = MMM_MAKE_STAT (MMM_MODULE_END)
+} MMM_STATS;
+
+int memmon_add_stat (THREAD_ENTRY *p_thread, MMM_STATS stat, uint64_t size);
+int memmon_sub_stat (THREAD_ENTRY *p_thread, MMM_STATS stat, uint64_t size);
+int memmon_move_stat (THREAD_ENTRY *p_thread, MMM_STATS before_stat, MMM_STATS after_stat, uint64_t size);
+int memmon_resize_stat (THREAD_ENTRY *p_thread, MMM_STATS stat, uint64_t before_size, uint64_t after_size);
+
+namespace cubperf
+{
+  class MMM;
```

Comment:
The upper-case class name like MMM and combined upper and lower-case class name like MMM_printer are not familiar. So, I recommand to use lower-case class name. (Or upper-case in case of typedef)

Reply:
I changed all upper-case naming to lower-case.